File: backend/app/schemas/audit.py
import uuid
from datetime import datetime

from pydantic import BaseModel, ConfigDict


class AuditLogOut(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: uuid.UUID
    created_at: datetime
    actor_user_id: uuid.UUID
    actor_email: str
    action: str
    target_type: str
    target_id: uuid.UUID
    target_label: str
    details: dict[str, object]


class AuditLogPage(BaseModel):
    items: list[AuditLogOut]
    total: int
    limit: int
    offset: int

File: frontend/src/types/audit.ts
export interface AuditLogOut {
  id: string;
  created_at: string;
  actor_user_id: string;
  actor_email: string;
  action: string;
  target_type: string;
  target_id: string;
  target_label: string;
  details: Record<string, unknown>;
}

export interface AuditLogPage {
  items: AuditLogOut[];
  total: number;
  limit: number;
  offset: number;
}

export const ACTION_LABELS: Record<string, string> = {
  "teacher.created": "Tạo giáo viên",
  "teacher.updated": "Cập nhật giáo viên",
  "teacher.activated": "Mở tài khoản",
  "teacher.deactivated": "Khóa tài khoản",
  "teacher.password_reset": "Đặt lại mật khẩu",
};
